crates/next-core/src/app_segment_config.rs

```diff
@@ -470,20 +471,33 @@ fn parse_config_value(
 pub async fn parse_segment_config_from_loader_tree(
     loader_tree: Vc<LoaderTree>,
 ) -> Result<Vc<NextSegmentConfig>> {
-    let loader_tree = loader_tree.await?;
-    let components = loader_tree.components.await?;
+    let loader_tree = &*loader_tree.await?;
+
+    parse_segment_config_from_loader_tree_internal(loader_tree).await
+}
+
+#[async_recursion]
+pub async fn parse_segment_config_from_loader_tree_internal(
+    loader_tree: &LoaderTree,
+) -> Result<Vc<NextSegmentConfig>> {
```

Comment:
Commenting here because github won't let me comment on the return line since it's not technically part of the diff... 😡 :octocat:

Should this `internal` function call `config.cell()`? It seems to me like it could just directly return the config, and then only the top-level call in `parse_segment_config_from_loader_tree` needs to construct a cell.

Otherwise you're constructing intermediate cells that must be cached, but are never used.